### PR TITLE
add term jump option

### PR DESCRIPTION
**Pull request by ldelossa**
This PR adds an option to disable jumping to a particular buffer on term command exit.

### CODE REVIEW

**Comment by ldelossa** [comment]
Hey @bhcleek 

> Why should the behaviour of jumping to errors be configurable for terminals differently than when a terminal isn't used?

You may have a point here, honestly I only use terminal when using vim-go.

> Why isn't the bang attribute handling sufficient?
What should bang accomplish here? I was copying the code used for "ToggleTermClose" pr I did before. 

> If we do go forward with this, then the documentation this PR introduces will need to be modified to make it clear that it controls jumping to the location of the first error in the error list (instead of saying that it controls jumping to the terminal window output).

sounds good to me.

File: autoload/go/term.vim [hunk 1]
```diff
@@ -204,7 +203,11 @@ func s:handle_exit(job_id, exit_status, state) abort
   endif
 
   call win_gotoid(a:state.winid)
-  call go#list#JumpToFirst(l:listtype)
+
+  if go#config#TermJumpOnExit()
+    call go#list#Window(l:listtype, len(l:errors))
```
[the remark this answers is not in the document]
Comment (ldelossa) [reply autoload/go/term.vim:208]: This prevents the location list from being opened all together. 
My rationale being if you don't want to jump, you don't want the location window to appear until you toggle it visible yourself. 
Okay with changing that rationale if you do not agree.

[the remark this answers is not in the document]
Comment (bhcleek) [reply autoload/go/term.vim:208]: That would be diverge from the (currently consistent) behavior of vim-go commands. I'd prefer not to diverge.

**Comment by bhcleek** [comment]
> You may have a point here, honestly I only use terminal when using vim-go.

I think you can get the behavior you want already by using the bang form of the commands that open a terminal. e.g. `:GoTest!`.

**Comment by bhcleek** [comment]
> What should bang accomplish here? I was copying the code used for "ToggleTermClose" pr I did before.

The bang prevents jumping to the first error (e.g. see `:help :GoTest`)

**Comment by ldelossa** [comment]
I think you are right. I neglected the "!" postfix on commands.

**Pull request closed without merging**